app/src/components/Modals/add-file-data-modal.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { FC, useEffect, useState } from "react";
+import {
+  Box,
+  Button,
+  Checkbox,
+  Divider,
+  FormControl,
+  FormLabel,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useToast,
+} from "@chakra-ui/react";
+import { AddFileIcon } from "../icons";
+import DropdownSelectInput from "../dropdown-select-input";
+import { InfoIcon, InfoOutlineIcon } from "@chakra-ui/icons";
+import { SubSectorWithRelations } from "@/app/[lng]/data/[step]/types";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useDispatch, useSelector } from "react-redux";
+import { addFileData, clear } from "@/features/city/fileDataSlice";
+import { RootState } from "@/lib/store";
+import { TFunction } from "i18next";
+
+interface AddFileDataModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  subsectors: SubSectorWithRelations[] | null;
+  t: TFunction;
+}
+
+export interface FileData {
+  subsectors: string;
+  scopes: string;
+}
+
+const AddFileDataModal: FC<AddFileDataModalProps> = ({
+  isOpen,
+  onClose,
+  subsectors,
+  t,
+}) => {
+  const scopes = [
+    {
+      value: 1,
+    },
+    {
+      value: 2,
+    },
+    {
+      value: 3,
+    },
+  ];
+
+  const [selectedScopes, setSelectedScopes] = useState<number[]>([]);
+
+  const handleSelectedScopes = (value: number, checked: boolean) => {
+    if (checked) {
+      setSelectedScopes([...selectedScopes, value]);
+    } else {
+      setSelectedScopes(selectedScopes.filter((scope) => scope !== value));
+    }
+  };
+  const dispatch = useDispatch();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    formState: { errors },
+  } = useForm<FileData>();
+
+  const onSubmit: SubmitHandler<FileData> = (data) => {
+    const scopeValues = selectedScopes.slice().join(",");
```

Comment:
Don't use CSV when you can use arrays or relations (Postgres has array columns that can be used with sequelize, and you can represent it as an array of values in JS instead of passing CSV strings around.